src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -20,6 +21,19 @@ public class DataAnnotationsMetadataProvider :
         IDisplayMetadataProvider,
         IValidationMetadataProvider
     {
+        private IStringLocalizerFactory _stringLocalizerFactory { get; set; }
+
+        public DataAnnotationsMetadataProvider()
```

Comment:
Shouldn't all creators of this `.Internal` type explicitly pass `null` or an `IStringLocalizerFactory`?